src/webpage.cpp

```diff
@@ -14,6 +17,19 @@ bool WebPage::acceptNavigationRequest(const QUrl &url, QWebEnginePage::Navigatio
         QDesktopServices::openUrl(url);
         return false;
     }
+    if (url.url().endsWith(".epub")) {
```

Comment:
It seems that you don't even have to launch the download yourself.

Qt starts the download itself.
You have to display the file selection dialog box in `startDownload` and set the path with `download->setPath`.

And be careful to refuse the download if the user "cancel" the dialog box. Else Qt still download the file using the default path.

Reply:
And it would work also on other files than `epub`.